### PR TITLE
cppclient: add unit test of getNodeInfo()

### PR DESCRIPTION
**Pull request by howjmay**
Imitate the unit test of getBalances(), and there is another unit test way which is imitated from iota.lib.cpp.

### CODE REVIEW

**Comment by howjmay** [comment]
@jserv  I fixed it

File: cppclient/tests/api_json.cc [hunk 1]
```diff
@@ -62,5 +62,38 @@ TEST_F(IotaJsonAPITest, GetBalances) {
   EXPECT_EQ(api.request, req);
   EXPECT_EQ(response, expected);
 }
+TEST_F(IotaJsonAPITest, GetNodeInfo) {
```
Comment (jserv) [inline cppclient/tests/api_json.cc]: There should be one blank line to split `GetNodeInfo` item and the previous one.

File: cppclient/tests/api_json.cc [hunk 2]
```diff
@@ -62,5 +62,38 @@ TEST_F(IotaJsonAPITest, GetBalances) {
...
+  MockAPI api;
+
+  json req;
+  req["command"] = "getNodeInfo";
+  json res;
+  char *str = "VBVEUQYE99LFWHDZRFKTGFHYGDFEAMAEBGUBTTJRFKHCFBRTXFAJQ9XIUEZQCJOQTZNOOHKUQIKOY9999";
```
Comment (jserv) [inline cppclient/tests/api_json.cc]: Avoid C-style string. Instead, make use of C++ `std::string`, which provides abstractions along with better memory management.

File: cppclient/tests/api_json.cc [hunk 3]
```diff
@@ -62,5 +62,38 @@ TEST_F(IotaJsonAPITest, GetBalances) {
...
+  auto response = api.getNodeInfo();
+  
+  // Imitae the unit test in getBlances in entangled
+  EXPECT_EQ(api.request, req);
+  EXPECT_STREQ((*response).latestMilestone.c_str(), str);
+  EXPECT_GE((*response).latestMilestoneIndex, res["latestMilestoneIndex"]);
+  EXPECT_GE((*response).latestSolidMilestoneIndex, res["latestSolidSubtangleMilestoneIndex"]);
+
+
+  /*
```
Comment (jserv) [inline cppclient/tests/api_json.cc]: Don't leave the following dead code.

**Pull request closed without merging**